src/main/cli/settings.c

```diff
@@ -1556,8 +1557,14 @@ const clivalue_t valueTable[] = {
 #endif
 #endif
 #ifdef USE_MCO
+#ifdef STM32G4
+    { "mco_on_pa8",     VAR_UINT8  | HARDWARE_VALUE | MODE_LOOKUP, .config.lookup = { TABLE_OFF_ON }, PG_MCO_CONFIG, offsetof(mcoConfig_t, enabled[0]) },
```

Comment:
Shouldn't this be available to all MCU types that support MCO?

The way this is done at the moment it seems to be weird that we are only ever using either `enabled[0]` or `enabled[1]`, but never both.

Reply:
The original MCO feature was introduced to replace a need for an external crystal/resonator for MAX chips, and was implemented on F7s to output the 27MHz from MCO2 with PLLI2S statically configured for 27MHz generation (#6163). Then the same mod followed for F4 (#7941). In these implementations a provisioning for MCO(1) was also made, hence the `enabled[2]` array.

On G4, there is only one MCO is supported (MCO(1)), and to generate 27MHz, it is necessary to use 27MHz HSE input with MCO source being HSE and scaler being 1. (The original PR had these parameters in the PG reset values.)

I decided to keep the array of 2 (`enabled[2]`) design to (1) avoid modifying existing PG to be modified, and (2) extended the notion to source and scaler to avoid confusion of a single parameters used for MCO(1) and MCO2.

As source and scaler configurability and two MCO support for F4 and F7 by @ModdQuad are expected (#9346), so there is a reason to keep the array of 2 design as is.

Thoughts?

Reply:
I am surprised that more FC designers do not take advantage of the MCO feature  to clock the OSD, an oscillator is around 50 cents, and the margins are so low, but I was unable to find a single config of an F7 board using it.   

What I can say from my gamma board bring-up a month ago is that the MC0 output is not automatically enabled by software only.  That is if you buy a F722 chip from digikey,   program it (DFU using companion over usb) with BetaFlight compiled with the 'USE_MCO' flag, the clock does not appear on the pin.  I debugged it for a day and came to the conclusion that the code is correct.  (Actually its very good, the MCO is initialized first, before any SPI coms, etc)  I think the issue is that you need to use 'ST-Link Utility' mass program/erase util to set a flag in a config register to enable output of MCO to the pin.  I left it at that without testing as I only have one working gamma board and it does not have a debugger interface to the F7 to make testing easy.  (hand placing the gyro was problematic)

My beta FC has finally been shipped from MacroFab after a corona virus delay,  it should be here tomorrow.  It has a couple of clocking options, but (STM32F722) MCO2/PC9 is hooked up to provide an 8MHz clock to a STM32G071 chip OSD in one configuration.   I am on the fence on using it.  While it is the best and cheapest option, it has not really gained any traction,  and is not used by any other FC manufacturers.  The safe option is to use an oscillator like everybody else.  Then again I changed a couple of other things that are not currently supported,  I connected a free pin for a OSD vertical blank interrupt.  The BF OSD runs off of a 60Hz timer, why not use a 59.97Hz  IRQ to synchronize everything. 

I will know a lot more about this in the next few days during bring-up.  I included a debugging interface for the F7 on this one, lesson learned.

Reply:
@jflyper: Ok, that makes sense. But in this case I think we should not worry too much about backwards compatibility of the parameter group and just increment the version. Instead, we should re-shuffle the parameter group to be an array type parameter group, or at least a parameter group containing an array of a struct - having a parameter group containing a list of parameter arrays to describe the same entities seems to be awkward.
@ModdQuad: That seems to be a major drawback - theoretically it should be possible to add support for changing the config register contents to Betaflight configurator, but especially with Unified Targets it would be hard to find a solution that only sets this flag on boards that need it, and I am not sure of potential negative implications of setting it on all boards.

Reply:
My sample size of one is pretty inconclusive, but there should be a mcu specific software fix if it exists at all.   I will let you know what I find out in the next couple of days.

Reply:
Ok, Ignore most of above, I now have a SWD port,  the 'USE_MCO' flag works as expected, the only issue I saw was that the 27MHz clock was not going all the way to gnd and would need some buffering, awfully fast for a pin though.  It works good at 8MHz.

The change I need is to BOTH the MUX and the DIVIDER to get an 8MHz passthrough clock.   Here is my change(hack) I made to line 42 in file https://github.com/betaflight/betaflight/blob/master/src/main/drivers/mco.c

Original:
        HAL_RCC_MCOConfig(RCC_MCO2, RCC_MCO2SOURCE_PLLI2SCLK, RCC_MCODIV_4);
Modified for 8MHz Output.
        HAL_RCC_MCOConfig(RCC_MCO2, RCC_MCO2SOURCE_HSE, RCC_MCODIV_1);
 
The MCO1 pin has a timer so it is more valuable during layout.